### PR TITLE
Add support for Glue registry

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Add support for Glue Schema registry (https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html)
1. Introduced registry_type under schema config to support Amazon Glue and Confluent schema registries
2. Refactored code to move all authentication codes to the new introduced KafkaSourceSecurityConfigurer
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [X ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/SchemaRegistryType.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+package org.opensearch.dataprepper.plugins.kafka.configuration;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Map;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public enum SchemaRegistryType {
+    GLUE("glue"),
```
Comment (hshardeesi) [inline data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/SchemaRegistryType.java:15]: nit: AWS_GLUE might be better naem

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSourceSecurityConfigurer.java [hunk 2]
```diff
@@ -119,25 +154,130 @@ public static void setOauthProperties(final KafkaSourceConfig kafkaSourConfig,
...
+            properties.put(SASL_JAAS_CONFIG,
                     "software.amazon.msk.auth.iam.IAMLoginModule required;");
         }
     }
+
+    public static String getBootStrapServersForMsk(final AwsIamAuthConfig awsIamAuthConfig, final AwsConfig awsConfig) {
+        AwsCredentialsProvider credentialsProvider = DefaultCredentialsProvider.create();
+        if (awsIamAuthConfig == AwsIamAuthConfig.ROLE) {
+            String sessionName = "data-prepper-kafka-session" + UUID.randomUUID();
+            StsClient stsClient = StsClient.builder()
```
Comment (hshardeesi) [inline data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSourceSecurityConfigurer.java:180]: I thought we have common way of getting credentials now. check with dlv@

**Review by hshardeesi — APPROVED** [review]
Few minor comments

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/SchemaRegistryType.java (hunk 1 above)
Comment (kkondaka) [reply data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/SchemaRegistryType.java:15]: Agreed. Will do in the next PR.

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSourceSecurityConfigurer.java (hunk 2 above)
Comment (kkondaka) [reply data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSourceSecurityConfigurer.java:180]: Yeah. We have to do that later.

File: data-prepper-plugins/kafka-plugins/README.md
Comment (asifsmohammed) [inline data-prepper-plugins/kafka-plugins/README.md:1]: Adding links to Glue and Kafka documentation will help users who want to try these out. And also the Kafka registry you added in description.

**Review by asifsmohammed — APPROVED** [review]

**Pull request merged**